### PR TITLE
Update go-address for latest SubnetID version

### LINKED ISSUE #220: Upgrade to latest version of go-address
There's a new version of `go-address` including a new implementation of `SubnetID` and slight modifications to the hierarchical address in order to adhere to the spec and make them compatible with their Rust implementation.

See:
  - Rust: https://github.com/adlrocha/ref-fvm/pull/4
  - Go: https://github.com/adlrocha/go-address/pull/1

### PR DESCRIPTION
**Pull request by adlrocha**
Fixes https://github.com/filecoin-project/eudico/issues/220

### CODE REVIEW

**Comment by adlrocha** [comment]
@dnkolegov, I am running out of battery in the airport, if you don't mind having a look at the linter errors I'd appreciate it. If you don't have time no worries, I'll check them out when I land. Thanks :pray:

File: chain/consensus/common/cns_validations.go [hunk 1]
```diff
@@ -494,7 +498,11 @@ func FilterBlockMessages(
 	// If subnet manager is not set we are in the root chain and we don't need to get parentSCA
 	// state
 	if submgr != nil {
-		parentSCA, pstore, err = crossmsg.GetSCAState(ctx, sm, submgr, netName.Parent(), baseTs)
+		p, err := netName.GetParent()
+		if err != nil {
+			log.Infof("Failed to get parent: %s", err)
```
Comment (dnkolegov) [inline chain/consensus/common/cns_validations.go]: Why not log.Error?

File: chain/consensus/common/cns_validations.go [hunk 2]
```diff
@@ -494,7 +498,11 @@ func FilterBlockMessages(
...
+		}
+		parentSCA, pstore, err = crossmsg.GetSCAState(ctx, sm, submgr, p, baseTs)
 		if err != nil {
 			log.Infof("Failed to get parent SCA state: %s", err)
```
Comment (dnkolegov) [inline chain/consensus/common/cns_validations.go]: Why not log.Error?

File: chain/consensus/delegcns/delegated.go [hunk 3]
```diff
@@ -83,6 +83,7 @@ func NewDelegatedConsensus(
 	genesis chain.Genesis,
 	netName dtypes.NetworkName,
 ) consensus.Consensus {
+	sn, _ := address.SubnetIDFromString(string(netName))
```
Comment (dnkolegov) [inline chain/consensus/delegcns/delegated.go:86]: Why error is ignored here?

File: chain/consensus/hierarchical/actors/sca/sca_actor.go [hunk 4]
```diff
@@ -424,7 +429,8 @@ func (a SubnetCoordActor) Fund(rt runtime.Runtime, params *SubnetIDParam) *abi.E
 	// Increment stake locked for subnet.
 	var st SCAState
 	rt.StateTransaction(&st, func() {
-		msg := fundMsg(rt, address.SubnetID(params.ID), secpAddr, value)
+		sn, _ := address.SubnetIDFromString(params.ID)
```
Comment (dnkolegov) [inline chain/consensus/hierarchical/actors/sca/sca_actor.go]: Why error is ignored here?

File: chain/consensus/hierarchical/actors/sca/sca_cross.go [hunk 5]
```diff
@@ -234,9 +236,10 @@ func incrementNonce(rt runtime.Runtime, nonceCounter *uint64) {
 func (st *SCAState) aggChildMsgMeta(rt runtime.Runtime, ch *schema.Checkpoint, aux map[string][]schema.CrossMsgMeta) {
 	for to, mm := range aux {
 		// Get the cid of MsgMeta from this subnet (if any)
-		metaIndex, msgMeta := ch.CrossMsgMeta(st.NetworkName, address.SubnetID(to))
+		sto, _ := address.SubnetIDFromString(to)
```
Comment (dnkolegov) [inline chain/consensus/hierarchical/actors/sca/sca_cross.go]: Why error is ignored here?

File: chain/consensus/hierarchical/actors/sca/sca_state.go [hunk 6]
```diff
@@ -104,9 +104,9 @@ func ConstructSCAState(store adt.Store, params *ConstructorParams) (*SCAState, e
 	if checkpointPeriod < minCheckpointPeriod {
 		checkpointPeriod = minCheckpointPeriod
 	}
-
+	sn, _ := address.SubnetIDFromString(params.NetworkName)
```
Comment (dnkolegov) [inline chain/consensus/hierarchical/actors/sca/sca_state.go]: Why error is ignored here?

File: chain/consensus/hierarchical/actors/sca/sca_subnet.go [hunk 7]
```diff
@@ -60,7 +60,8 @@ func fundMsg(rt runtime.Runtime, id address.SubnetID, secp address.Address, valu
 	// Transform To and From to HCAddresses
 	to, err := address.NewHCAddress(id, secp)
 	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to create HCAddress")
-	from, err := address.NewHCAddress(id.Parent(), secp)
+	p, _ := id.GetParent()
```
Comment (dnkolegov) [inline chain/consensus/hierarchical/actors/sca/sca_subnet.go:63]: Why error is ignored here?

**Review by dnkolegov — APPROVED** [review]
LGTM. Just several questions about ignoring returned errors and logging messages.

**Pull request merged**